src/test/java/io/debezium/connector/vitess/VitessConnectorIT.java

```diff
@@ -290,22 +290,101 @@ public void shouldUsePrevVgtidAsOffsetWhenNoVgtidInGrpcResponse() throws Excepti
 
     @Test
     public void shouldMultiShardSubscriptionHaveMultiShardGtidsInVgtid() throws Exception {
+        final boolean hasMultipleShards = true;
+
         TestHelper.executeDDL("vitess_create_tables.ddl", TestHelper.TEST_SHARDED_KEYSPACE);
         TestHelper.applyVSchema("vitess_vschema.json");
-        startConnector(true);
+        startConnector(hasMultipleShards);
         assertConnectorIsRunning();
 
-        // insert 1 row
         int expectedRecordsCount = 1;
         consumer = testConsumer(expectedRecordsCount);
-        TableId table = tableIdFromInsertStmt(INSERT_NUMERIC_TYPES_STMT, TestHelper.TEST_SHARDED_KEYSPACE);
-        executeAndWait(INSERT_NUMERIC_TYPES_STMT, TestHelper.TEST_SHARDED_KEYSPACE);
+        assertInsert(INSERT_NUMERIC_TYPES_STMT, schemasAndValuesForNumericTypes(), TestHelper.TEST_SHARDED_KEYSPACE, TestHelper.PK_FIELD, hasMultipleShards);
 
-        SourceRecord record = assertRecordInserted(topicNameFromInsertStmt(INSERT_NUMERIC_TYPES_STMT, TestHelper.TEST_SHARDED_KEYSPACE));
-        // verify the offset has multiple shard gtids
-        assertRecordOffset(record, RecordOffset.fromSourceInfo(record), true);
+        stopConnector();
+        assertConnectorNotRunning();
+    }
+
+    @Test
+    public void shouldUseMultiColumnPkAsRecordKey() throws Exception {
```

Comment:
Could you add the `@FixFor` annotation?

Reply:
That's a good suggestion! `@FixFor` added to all the test cases in this PR. I'll remember to do it systematically :)